src/hosted-buttons/utils.js

```diff
@@ -0,0 +1,157 @@
+/* @flow */
+
+import { request, memoize, popup } from "@krakenjs/belter/src";
+import { getSDKHost, getClientID } from "@paypal/sdk-client/src";
+
+import { DEFAULT_POPUP_SIZE } from "../zoid/checkout";
+
+import type {
+  ButtonVariables,
+  CreateAccessToken,
+  CreateOrder,
+  GetCallbackProps,
+  HostedButtonDetailsParams,
+  OnApprove,
+  RenderForm,
+} from "./types";
+
+const entryPoint = "SDK";
+const baseUrl = `https://${getSDKHost()}`;
+const apiUrl = baseUrl.replace("www", "api");
+
+const getHeaders = (accessToken?: string) => ({
+  ...(accessToken && { Authorization: `Bearer ${accessToken}` }),
+  "Content-Type": "application/json",
+  "PayPal-Entry-Point": entryPoint,
+});
+
+export const createAccessToken: CreateAccessToken = memoize<CreateAccessToken>(
+  (clientId) => {
+    return request({
+      url: `${apiUrl}/v1/oauth2/token`,
+      method: "POST",
+      body: "grant_type=client_credentials",
+      headers: {
+        Authorization: `Basic ${btoa(clientId)}`,
+        "Content-Type": "application/json",
+      },
+    }).then((response) => response.body.access_token);
+  }
+);
+
+const getButtonVariable = (variables: ButtonVariables, key: string): string =>
+  variables?.find((variable) => variable.name === key)?.value ?? "";
+
+const getFundingSource = (paymentSource) => {
+  if (paymentSource === "credit") {
```

Comment:
Maybe we could import these values from `FUNDING` in `@paypal/sdk-constants/src` instead of hardcoding `"credit"` and `'CARD'`?

Reply:
yes! I like it 72e75b05